### DISCUSSION TITLE
How to make list take full height

### DISCUSSION THREAD

**Discussion by zeljkosh** [fyne-io/fyne#3225] [Q&A] [answered]
i created window and added some items, to fully see how i will paste code underneath.

`hBox := container.NewHBox(layout.NewSpacer(), button, layout.NewSpacer(), btn)
hBox2 := container.NewVBox(factText)
vBox := container.NewVBox(title, hBox, widget.NewSeparator(), hBox2, listMain)
win.SetContent(vBox)`

But my list is only about 30 pixel high, even though it does have space for much more, so basicly i see only one item, and i must scroll to see other ones.
Is the list height decided by the parent, list itself or the lists child ?
![Screenshot from 2022-08-26 14-28-38](https://user-images.githubusercontent.com/29103626/186903611-8396288a-ed97-41b1-bfe5-5d00739289df.png)
?

**Comment by Jacalz** [comment]
You should avoid using the VBox container as that collapses the contents to their vertical MinSize. I would suggest that you use a `container.NewBorder()` instead and set the list as the center element and the items above the list in the `top` position.

**Reply by zeljkosh** [reply]
Hey, thanks for the help. 
Border does have this thing where center always tryes to take the maximum size, which is okay, but for example i want the size for the left to always be lets say 300 width.
My left is in this case a label, when i click a button labels text is changed, and label has TextWrapBreak enabled,
i use resize to set starting size of the label, but when the new text is added size of label falls to one letter.
How can i set minimum width of the left container ?
I would like following layout, 
top is whole width and height is max of 30,
left is whole height and max width of 300
center always takes the rest.
`left := container.NewVBox(factText)

	factText.Wrapping = fyne.TextWrapBreak

	hBox := container.New(layout.NewHBoxLayout(), layout.NewSpacer(), button, layout.NewSpacer(), btn)

	//vBox := container.NewVBox(title, hBox, widget.NewSeparator(), hBox2)
	content := container.New(layout.NewBorderLayout(hBox, nil, left, nil),
		hBox, left, listMain)
	win.SetContent(content)
	win.Resize(fyne.NewSize(1000, 600))
	win.ShowAndRun()
}`

**Comment by andydotxyz** [comment, marked as answer]
Might be useful to read more about layouts at https://developer.fyne.io/faq/layout

**Comment by Jacalz** [comment]
I would suggest watching https://m.youtube.com/watch?v=LWn1403gY9E&feature=emb_title as well.
